doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -15,46 +12,63 @@ feature, and submitting pull requests.
 
 .. note:: A `contributor license agreement`_ must be signed before any contributing pull requests can be merged.
 
-Developer Workflow
-------------------
-
 Overview:
 
-#. `Discuss design changes or big development efforts`_
-#. `Get set up`_
-#. `Create new branch`_
-#. `Develop and test contributions locally`_
+#. `Choosing a task`_
+#. `Development`_
+
+    #. `Get set up`_
+    #. `Create new branch`_
+    #. `Develop and test contributions locally`_
+
+       #. `Add new tests`_
+
+#. `Design`_
 
-   #. `Add new tests`_
+    #. `Creating a design issue`_
+    #. `Leading a design discussion`_
 
-#. `Request feedback on your changes`_
+#. `Contributing changes`_
 
    #. `Push your work to your feature branch`_
    #. `Ask for feedback on your branch early (optional)`_
    #. `Submit pull request`_
    #. `Find a reviewer`_
    #. `Work with your reviewers`_
+   #. `Before merging`_
+   #. `After merging`_
+
+#. `Other useful information`_
+
+   #. `Chapel git workflow`_
+   #. `Policy details`_
+
+
 
-#. `Get ready to merge`_
-#. `Watch automatic testing and address issues`_
+.. _Choosing a task
 
-`HOWTO and Git details`_
+Choosing a task
+---------------
 
-`Policy details`_
+If you do not already know what task to work on, the `Contributing`_ page has
+some tips for finding a task.
 
-.. _Discuss design changes or big development efforts:
+It is important to communicate with other people who might be working on the
+same area. For small tasks, this can be as simple as commenting on an issue.
+For larger projects, you should discuss the approach in a design issue on
+GitHub. See the `Design`_ section for guidance on creating a design issue.
```

Comment:
I am a bit confused by this paragraph.

"commenting on an issue" for small tasks seem a bit vague, for larger project we seem to suggest opening a new design issue. But "commenting on an issue" to show interest for a big project can be fine, too.

Maybe after the first sentence we can just say that the contriubtor should comment under the relevant issue (without separating big and small tasks just yet) and even suggest that they should maybe give a heads-up on Gitter channel if there is no issue.

Reply:
I guess this was assuming the cases of:

1. An issue exists and user would like to work on it, so they comment on it. Any design discussion can take place there.
2. A user has a big project idea, but no issue exists - they should open an issue to discuss it before pursuing it

Sounds like you're suggesting:

1. If an issue exists, comment on it
2. If an issue does not exist, create it (and see [design] section)

where the size of the project does not matter.

>  maybe give a heads-up on Gitter channel if there is no issue.

I am not sure I would consider this a requirement.

Reply:
> 1. If an issue exists, comment on it
> 2. If an issue does not exist, create it (and see [design] section)
> 
> where the size of the project does not matter.

Yes, I think this is a better dichotomy than small/big, which are open to interpretation.

I agree that bringing things up in Gitter shouldn't be a requirement. So, I am fine if you leave it out.

Reply:
I feel like this doesn't cover the case for when there's a vague feature request and someone jumps on it by commenting "I'll work on this".

How about this?

1. If an issue exists, comment on it
   - If that's a big or complicated task, see [design] section for guidance on how to proceed
2.   If an issue does not exist, create one and see [design] section for guidance

Reply:
That's a good point. What you suggest sounds good to me.